dhis-2/dhis-api/src/main/java/org/hisp/dhis/setting/SystemSettings.java

```diff
@@ -745,4 +745,15 @@ default boolean isHideUnapprovedDataInAnalytics() {
     // -1 means approval is disabled
     return getIgnoreAnalyticsApprovalYearThreshold() >= 0;
   }
+
+  /**
+   * @return A regex pattern string that enforces the current password validation rules
+   */
+  default String getPasswordValidationPattern() {
+    return asString(
+        "passwordValidationPattern",
+        String.format(
+            "^(?=.*[A-Z])(?=.*[a-z])(?=.*\\d)(?=.*[\\W_])[A-Za-z\\d\\W_]{%d,%d}$",
```

Comment:
That regex seems too specific. With this a upper case letter must always be before a lower case one, both before a digit and all of them before a non-word character. All of that must be followed by the min-max number of letters, gitis, and non-word characters. That seems illogical. Why use `.*` in several places allowing anything but then at the end be so specific? I would assume `\W` basically excludes what you then included again with `A-Za-z\\d` so effectively that will be very close or similar to `.`. 

I think what you wanted to say is something entirely different where it is min-max number of characters for which at least one is a lower letter, one an upper letter, one a digit, and one a special character.

Reply:
For example: `+1aaaaAA00` does not match. That has 8 letters but as I explained you basically ask for 4+min-max. But considering that, even `+1aaaaAA0012345676` would not match.